src/TextInput.tsx

```diff
@@ -1,25 +1,34 @@
+import {ForwardRefComponent as PolymorphicForwardRefComponent} from '@radix-ui/react-polymorphic'
 import classnames from 'classnames'
 import React from 'react'
-import {ComponentProps, Merge} from './utils/types'
+import {Merge} from './utils/types'
+import TextInputWrapper, {StyledWrapperProps} from './_TextInputWrapper'
 import UnstyledTextInput from './_UnstyledTextInput'
-import TextInputWrapper from './_TextInputWrapper'
 
 type NonPassthroughProps = {
-  className?: string
   /** @deprecated Use `leadingVisual` or `trailingVisual` prop instead */
   icon?: React.ComponentType<{className?: string}>
   leadingVisual?: string | React.ComponentType<{className?: string}>
   trailingVisual?: string | React.ComponentType<{className?: string}>
 } & Pick<
-  ComponentProps<typeof TextInputWrapper>,
-  'block' | 'contrast' | 'disabled' | 'monospace' | 'sx' | 'width' | 'maxWidth' | 'minWidth' | 'variant' | 'size'
+  StyledWrapperProps,
+  | 'block'
+  | 'contrast'
+  | 'disabled'
+  | 'monospace'
+  | 'sx'
+  | 'width'
+  | 'maxWidth'
+  | 'minWidth'
+  | 'variant'
+  | 'size'
+  | 'validationStatus'
 >
 
-// Note: using ComponentProps instead of ComponentPropsWithoutRef here would cause a type issue where `css` is a required prop.
-type TextInputInternalProps = Merge<React.ComponentPropsWithoutRef<typeof UnstyledTextInput>, NonPassthroughProps>
```

Comment:
This line was the root of the problem. Passing a [styled](https://github.com/styled-components/styled-components) component type to [`React.ComponentPropsWithoutRef`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/190ce439349e1fe23fd467f3f76d66be80d12fa0/types/react/index.d.ts#L845) yields an `any` type

```ts
import styled from 'styled-components'
import React from 'react'

const StyledComponent = styled.div`...`

type Props = 
  React.ComponentPropsWithoutRef<typeof StyledComponent>
// any
```

Reply:
Good to know so we don't make this mistake in the future.